spyder/widgets/findreplace.py

```diff
@@ -400,7 +417,18 @@ def find(self, changed=True, forward=True,
             regexp = self.re_button.isChecked()
             found = self.editor.find_text(text, changed, forward, case=case,
                                           words=words, regexp=regexp)
-            self.search_text.lineEdit().setStyleSheet(self.STYLE[found])
+
+            if not found and regexp and rexexp_error_msg(text) is not None:
+                stylesheet = self.STYLE['regexp_error']
+                tooltip = self.TOOLTIP['regexp_error']
+                error_msg = rexexp_error_msg(text)
+                if error_msg:
+                    tooltip += ': ' + error_msg
+            else:
+                stylesheet = self.STYLE[found]
+                tooltip = self.TOOLTIP[found]
+            self.search_text.lineEdit().setStyleSheet(stylesheet)
+            self.search_text.setToolTip(tooltip)

```

Comment:
This code could be refactored, It has repeated calls to `rexexp_error_msg`, also both "if" are checking for the error message:
```
            stylesheet = self.STYLE[found]
            tooltip = self.TOOLTIP[found]
            if not found and regexp:
                error_msg = rexexp_error_msg(text)    
                if error_msg:
                    stylesheet = self.STYLE['regexp_error']
                    tooltip = "{}: {}".format(self.TOOLTIP['regexp_error'], error_msg)

            self.search_text.lineEdit().setStyleSheet(stylesheet)
            self.search_text.setToolTip(tooltip)
```

Reply:
Thanks that's a more readable version (Note that calling `rexexp_error_msg()` twice is not a big deal in itself since `re.compile()` is cached).

I've applied your suggestion with the exception of not switching the string formatting. IMHO `"{}: {}".format(` is not making it more readable in this case (And also I don't want pep8 to moan again about the line length).